engine.py

```diff
@@ -637,6 +638,85 @@ def nullslast(tuple_list):
+            db_name = Transaction().database.name
+
+            image = '%s/%s/%s/%s/%s' % (
+                path,
+                db_name,
+                file_id[:2],
+                file_id[2:4:],
+                file_id,
+                )
+            if not os.path.isfile(image):
+                return False
+            mimetype = magic.from_file(image, mime=True)
+            if mimetype in IMAGE_TYPES:
+                return True
+            return False
+
+        def thumbnail(filename, size, crop=None, quality=85):
+            '''Create thumbnail image
+
+            :param filename: image digest - '2566a0e6538be8e094431ff46ae58950'
+            :param size: size thumb - '100x100'
+            :param crop: crop thumb - top, middle, bottom or None
+            :param quality: JPEG quality 1-100
+            :return file name
+
+            Example jinja2 tag:
+            <img src="file://{{ attachment.raw.file_id|thumbnail("200x200") }}"/>
+            '''
+            width, height = [int(x) for x in size.split('x')]
+
+            path = config.get('database', 'path')
+            db_name = Transaction().database.name
+
+            original_filename = os.path.join(path, db_name, filename[0:2], filename[2:4], filename)
+
+            size = (width, height)
+
+            try:
+                img = Image.open(original_filename)
+            except IOError:
+                return
+
+            if crop:
+                img_ratio = img.size[0] / float(img.size[1])
+                ratio = size[0] / float(size[1])
+
+                #The image is scaled/cropped vertically or horizontally depending on the ratio
+                if ratio > img_ratio:
+                    img = img.resize((size[0], size[0] * img.size[1] / img.size[0]), Image.ANTIALIAS)
+                    # Crop in the top, middle or bottom
+                    if crop == 'top':
+                        box = (0, 0, img.size[0], size[1])
+                    elif crop == 'bottom':
+                        box = (0, img.size[1] - size[1], img.size[0], img.size[1])
+                    else :
+                        box = (0, (img.size[1] - size[1]) / 2, img.size[0], (img.size[1] + size[1]) / 2)
+                    img = img.crop(box)
+                elif ratio < img_ratio:
+                    img = img.resize((size[1] * img.size[0] / img.size[1], size[1]), Image.ANTIALIAS)
+                    # Crop in the top, middle or bottom
+                    if crop == 'top':
+                        box = (0, 0, size[0], img.size[1])
+                    elif crop == 'bottom':
+                        box = (img.size[0] - size[0], 0, img.size[0], img.size[1])
+                    else :
+                        box = ((img.size[0] - size[0]) / 2, 0, (img.size[0] + size[0]) / 2, img.size[1])
+                    img = img.crop(box)
+            else:
+                img.thumbnail(size)
+
+            # save tmp file and return file name
```

Comment:
S'ha enfocat també generar el base64 en memoria, fins arribar al punt de weasyprint ens dona el missatge:

    weasyprint Failed to load image at Pixbuf error: Unrecognized image file format

El codi en que es desenvolupava és:

```
buffer = BytesIO()
img.save(buffer, img.format, quality=quality)
myimage = buffer.getvalue()
mimetype = img.get_format_mimetype()
import pdb; pdb.set_trace()
return "data:%s;base64,%s" % (mimetype, pybase64.b64encode(myimage))
```

Cal pensar també s'ha de mostrar imatges tif, que si no es per file:// tampoc es mostrarien